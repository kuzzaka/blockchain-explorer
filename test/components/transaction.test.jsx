import React from 'react';
import { mount, shallow } from 'enzyme';
import Transaction from '../../components/transaction';
import txData from '../stubs/tx-data';

test('Transaction', () => {
  const spy = jest.spyOn(Transaction.prototype, 'toggleWrap');
  const wrapper = mount(<Transaction data={txData[2]} />);
  expect(wrapper.find('article')).toExist();
  expect(wrapper.find('button')).toExist();
  wrapper.find('button').simulate('click');
  expect(spy).toHaveBeenCalled();
});

test('getValueString', () => {
  expect(Transaction.getValueString(txData[2].out[0])).toEqual('0.05640296 BTC');
  expect(Transaction.getValueString({})).toEqual('No value provided in BTC');
});

test('getTotalValueString', () => {
  expect(Transaction.getTotalValueString(txData[2].out)).toEqual('Total transaction value - 0.28269165 BTC');
  expect(Transaction.getTotalValueString([])).toEqual('Total transaction value - 0 BTC');
});

test('with wrapper Transaction snapshot', () => {
  const wrapper = shallow(<Transaction data={txData[2]} />);
  expect(wrapper).toMatchSnapshot();
});

test('without wrapper Transaction snapshot', () => {
  const wrapper = shallow(<Transaction data={txData[0]} />);
  expect(wrapper).toMatchSnapshot();
});
