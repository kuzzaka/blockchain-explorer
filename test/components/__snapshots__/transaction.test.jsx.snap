// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with wrapper Transaction snapshot 1`] = `
<article
  className="jsx-4190627686"
>
  <header
    className="jsx-4190627686"
  >
    <h5
      className="jsx-4190627686"
    >
      460bf9d5db9f1df960c03b5225316ac77d1965b8f4ac429e0e66a6e097018a07
    </h5>
    <time
      className="jsx-4190627686"
      dateTime="Sat Dec 17 2016"
    >
      Sat Dec 17 2016
    </time>
  </header>
  <section
    className="jsx-4190627686 inputs"
  >
    <h6
      className="jsx-4190627686"
    >
      Inputs
    </h6>
    <div
      className="jsx-4190627686 address-block"
      key="3AmYWkoqFh7M8MTb5hHQAYj5RUUVuKRrUq4"
    >
      <p
        className="jsx-4190627686"
      >
        3AmYWkoqFh7M8MTb5hHQAYj5RUUVuKRrUq
      </p>
    </div>
  </section>
  <span
    aria-label="right arrow emoji"
    className="jsx-4190627686 arrow"
    role="img"
  >
    ➡️
  </span>
  <section
    className="jsx-4190627686 out"
  >
    <h6
      className="jsx-4190627686"
    >
      Outputs
    </h6>
    <div
      className="jsx-4190627686 address-block"
      key="115k5rUF99rtRErbEt7zpKESPKyfnhvVbK0"
    >
      <p
        className="jsx-4190627686"
      >
        115k5rUF99rtRErbEt7zpKESPKyfnhvVbK
      </p>
      <p
        className="jsx-4190627686"
      >
        5640296 BTC
      </p>
    </div>
    <div
      className="jsx-4190627686 address-block"
      key="3P5Ro2bjWPTQUv1mZznBPA9jVzb8z2oPug1"
    >
      <p
        className="jsx-4190627686"
      >
        3P5Ro2bjWPTQUv1mZznBPA9jVzb8z2oPug
      </p>
      <p
        className="jsx-4190627686"
      >
        4339691 BTC
      </p>
    </div>
    <div
      className="jsx-4190627686 address-block"
      key="3P5Ro2bjWPTQUv1mZznBPA9jVzb8z2oPug2"
    >
      <p
        className="jsx-4190627686"
      >
        3P5Ro2bjWPTQUv1mZznBPA9jVzb8z2oPug
      </p>
      <p
        className="jsx-4190627686"
      >
        4339691 BTC
      </p>
    </div>
    <div
      className="jsx-4190627686 address-block"
      key="3P5Ro2bjWPTQUv1mZznBPA9jVzb8z2oPug3"
    >
      <p
        className="jsx-4190627686"
      >
        3P5Ro2bjWPTQUv1mZznBPA9jVzb8z2oPug
      </p>
      <p
        className="jsx-4190627686"
      >
        4339691 BTC
      </p>
    </div>
  </section>
  <div
    className="jsx-4190627686 button-block"
  >
    <div
      className="jsx-4190627686 total-value"
    >
      Total transaction value - 18659369 BTC
    </div>
    <Button
      onClick={[Function]}
    >
      Show more
    </Button>
  </div>
  <JSXStyle
    css="article.jsx-4190627686{display:grid;border:black 2px solid;margin-bottom:.5rem;padding:1rem;grid-template-rows:1fr minmax(1fr,4fr) 1fr;grid-template-columns:1fr min-content 2fr;grid-template-areas: \\"a a a\\"  \\"b c d\\"  \\"e e e\\";grid-gap:.5rem;width:100%;}header.jsx-4190627686{padding:.5rem 1rem;font-weight:bolder;grid-area:a;}p.jsx-4190627686{padding:0 1rem;word-break:break-word;}p.jsx-4190627686+p.jsx-4190627686{padding-bottom:.5rem;}span.jsx-4190627686{-webkit-align-self:center;-ms-flex-item-align:center;align-self:center;}h6.jsx-4190627686{margin-left:1rem;}.inputs.jsx-4190627686,.out.jsx-4190627686{border:darkslategray 1px dashed;}.inputs.jsx-4190627686{grid-area:b;}.out.jsx-4190627686{grid-area:d;}.arrow.jsx-4190627686{grid-area:c;}.button-block.jsx-4190627686{grid-area:e;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row-reverse;-ms-flex-direction:row-reverse;flex-direction:row-reverse;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}.total-value.jsx-4190627686{margin-right:1rem;font-weight:bolder;-webkit-align-self:center;-ms-flex-item-align:center;align-self:center;}.address-block.jsx-4190627686{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}"
    styleId="4190627686"
  />
</article>
`;

exports[`without wrapper Transaction snapshot 1`] = `
<article
  className="jsx-4190627686"
>
  <header
    className="jsx-4190627686"
  >
    <h5
      className="jsx-4190627686"
    >
      f94528b2012446f4e367694871d07d5c726788e2f20d5553161a24debbec5118
    </h5>
    <time
      className="jsx-4190627686"
      dateTime="Sat Dec 17 2016"
    >
      Sat Dec 17 2016
    </time>
  </header>
  <section
    className="jsx-4190627686 inputs"
  >
    <h6
      className="jsx-4190627686"
    >
      Inputs
    </h6>
    <div
      className="jsx-4190627686 address-block"
      key="NaN"
    >
      <p
        className="jsx-4190627686"
      >
        No address provided
      </p>
    </div>
  </section>
  <span
    aria-label="right arrow emoji"
    className="jsx-4190627686 arrow"
    role="img"
  >
    ➡️
  </span>
  <section
    className="jsx-4190627686 out"
  >
    <h6
      className="jsx-4190627686"
    >
      Outputs
    </h6>
    <div
      className="jsx-4190627686 address-block"
      key="1CK6KHY6MHgYvmRQ4PAafKYDrg1ejbH1cE0"
    >
      <p
        className="jsx-4190627686"
      >
        1CK6KHY6MHgYvmRQ4PAafKYDrg1ejbH1cE
      </p>
      <p
        className="jsx-4190627686"
      >
        1250503278 BTC
      </p>
    </div>
  </section>
  <div
    className="jsx-4190627686 button-block"
  >
    <div
      className="jsx-4190627686 total-value"
    >
      Total transaction value - 1250503278 BTC
    </div>
  </div>
  <JSXStyle
    css="article.jsx-4190627686{display:grid;border:black 2px solid;margin-bottom:.5rem;padding:1rem;grid-template-rows:1fr minmax(1fr,4fr) 1fr;grid-template-columns:1fr min-content 2fr;grid-template-areas: \\"a a a\\"  \\"b c d\\"  \\"e e e\\";grid-gap:.5rem;width:100%;}header.jsx-4190627686{padding:.5rem 1rem;font-weight:bolder;grid-area:a;}p.jsx-4190627686{padding:0 1rem;word-break:break-word;}p.jsx-4190627686+p.jsx-4190627686{padding-bottom:.5rem;}span.jsx-4190627686{-webkit-align-self:center;-ms-flex-item-align:center;align-self:center;}h6.jsx-4190627686{margin-left:1rem;}.inputs.jsx-4190627686,.out.jsx-4190627686{border:darkslategray 1px dashed;}.inputs.jsx-4190627686{grid-area:b;}.out.jsx-4190627686{grid-area:d;}.arrow.jsx-4190627686{grid-area:c;}.button-block.jsx-4190627686{grid-area:e;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row-reverse;-ms-flex-direction:row-reverse;flex-direction:row-reverse;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}.total-value.jsx-4190627686{margin-right:1rem;font-weight:bolder;-webkit-align-self:center;-ms-flex-item-align:center;align-self:center;}.address-block.jsx-4190627686{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}"
    styleId="4190627686"
  />
</article>
`;
